noxfile.py

```diff
@@ -58,81 +39,75 @@ def tests(session: nox.Session) -> None:
     session.run("npm", "run", "test")
 
 
-def _get_package_data(package):
-    json_uri = f"https://registry.npmjs.org/{package}"
-    with url_lib.urlopen(json_uri) as response:
-        return json.loads(response.read())
-
-
-def _update_npm_packages(session: nox.Session) -> None:
-    pinned = {
-        "vscode-languageclient",
-        "@types/vscode",
-        "@types/node",
-    }
-    package_json_path = pathlib.Path(__file__).parent / "package.json"
-    package_json = json.loads(package_json_path.read_text(encoding="utf-8"))
-
-    for package in package_json["dependencies"]:
-        if package not in pinned:
-            data = _get_package_data(package)
-            latest = "^" + data["dist-tags"]["latest"]
-            package_json["dependencies"][package] = latest
-
-    for package in package_json["devDependencies"]:
-        if package not in pinned:
-            data = _get_package_data(package)
-            latest = "^" + data["dist-tags"]["latest"]
-            package_json["devDependencies"][package] = latest
-
-    # Ensure engine matches the package
-    if (
-        package_json["engines"]["vscode"]
-        != package_json["devDependencies"]["@types/vscode"]
-    ):
-        print(
-            "Please check VS Code engine version and @types/vscode version in package.json."
-        )
-
-    new_package_json = json.dumps(package_json, indent=4)
-    # JSON dumps uses \n for line ending on all platforms by default
-    if not new_package_json.endswith("\n"):
-        new_package_json += "\n"
-    package_json_path.write_text(new_package_json, encoding="utf-8")
-
-    session.run("npm", "audit", "fix", external=True)
-    session.run("npm", "install", external=True)
-
-
-def _setup_template_environment(session: nox.Session) -> None:
-    session.install("wheel", "pip-tools")
-    _update_pip_packages(session)
-    _install_bundle(session)
-
-
-@nox.session(python="3.7")
+@nox.session()
 def install_bundled_libs(session):
     """Installs the libraries that will be bundled with the extension."""
     session.install("wheel")
-    _install_bundle(session)
+    session.install(
+        "-t",
+        "./bundled/libs",
+        "--no-cache-dir",
+        "--implementation",
+        "py",
+        "--no-deps",
```

Comment:
You also want to specify you only want wheels (so nothing gets built), and that you require hashes (to verify you're getting what you expect); see https://github.com/brettcannon/pip-secure-install/blob/92f400e3191171c1858cc0e0d9ac6320173fdb0c/action.yml#L22 for details:

```suggestion
        "--no-deps",
        "--require-hashes",
        "--only-binary",
        ":all:",
```